### ISSUE TITLE
Marvin is pretty verbose/spammy

### ISSUE THREAD

**Issue by Atemu** [timokau/marvin-mk2#108]
Currently, you need to enable Marvin via commands in the PR description or a separate comment when you forget to include it in the PR like I always do. It then adds a comment describing itself, adds two labels (separately but GH does batch them after a short while thankfully), requests a review from a suitable reviwer and then changes labels to reflect the status after its action.

You can find a block like this at the very top of pretty much all Marvin PRs:

![image](https://user-images.githubusercontent.com/18599032/103267897-e03c8100-49b2-11eb-9601-8587b2fc011c.png)

I think that is pretty verbose and could be simplified:

* The comment commands are nice but the boilerplate setup that needs to happen in every Marvin PR could also be implied by the PR owner setting the `marvin` label on their PR. Simple and unintrusive.
* needs_reviewer is implicit when marvin is first enabled (every PR needs at least one reviewer) and this is the only entry point AFAICT. Marvin adds this label only to remove again it seconds later if it finds a suitable reviwer.
* Marvin doesn't need to explain itself in every PR. If it's not immediately obvious what it does from its actions or if one wants to find out more, a user can simply click on Marvin's name.

I think it'd be better if the boilerplate was limited to:

* Owner sets `marvin` label
* Marvin requests review
* Marvin sets `awaiting_reviewer` label

**Comment by timokau** [comment]
Thanks for the feedback!

>     * The comment commands are nice but the boilerplate setup that needs to happen in every Marvin PR could also be implied by the PR owner setting the `marvin` label on their PR. Simple and unintrusive.

Unfortunately not everybody can set labels on their own PRs. The number of people that can is growing now that maintainer-team invites have picked up again, but that is still far from everybody.

>     * needs_reviewer is implicit when marvin is first enabled (every PR needs at least one reviewer) and this is the only entry point AFAICT.

Not necessarily, at least not as long as marvin is not enabled by default. A PR may already be in review, or the author may have already pinged a reviewer.

> Marvin adds this label only to remove again it seconds later if it finds a suitable reviwer.

Only in the lucky case where the `needs_reviewer` queue is empty. I think that is okay, I don't really consider it an issue to add/remove labels. It doesn't generate email notifications and it makes it easier to understand the state transitions.

>     * Marvin doesn't need to explain itself in every PR. If it's not immediately obvious what it does from its actions or if one wants to find out more, a user can simply click on Marvin's name.

This is a tradeoff between ease of use / discoverability and verbosity. You can discover the same information by clicking on the name, but that is a lot less obvious. There are a lot of bots on GitHub, and reviewers may just write it off as some labeling bot and ignore it. I agree that we should minimize unnecessary notifications, but at least for now I think the explanation is useful enough to justify the notification.

I suspect when/if this bot leaves the experimental phase and goes through an RFC, this will be discussed again.

> I think it'd be better if the boilerplate was limited to:
> 
>     * Owner sets `marvin` label
> 
>     * Marvin requests review
> 
>     * Marvin sets `awaiting_reviewer` label

If you prefer and have the necessary permission, you can already skip the commands and set the `marvin` and `needs_reviewer` labels manually. You might have to wait a bit until marvin does a triage run and discovers the new PR. Immediately requesting a review will likely be the exception rather than the rule, so I do not think we should optimize for that exceptional case.

To summarize: I do not consider label-changes problematic. As you said yourself, github will group them so I think the visual noise is acceptable and they do not generate notifications. I agree that notification-generating messages should be minimized, but I think that the introduction does more good than harm for now.

**Comment by Atemu** [comment]
> Unfortunately not everybody can set labels on their own PRs.

The commands would still work just fine in that case ;)

> A PR may already be in review, or the author may have already pinged a reviewer.

An author is only allowed to set needs_reviewer though, right? Also, Marvin could figure out existing reviwers and reviews.

> you can already skip the commands and set the marvin and needs_reviewer labels manually.

Oh! That's good to know, will do that from now on when I forget to call marvin.

**Comment by timokau** [comment]
> > Unfortunately not everybody can set labels on their own PRs.
> 
> The commands would still work just fine in that case ;)

Ah, I thought you wanted that to be the only way to interact with marvin. I feel like that might make marvin a little harder to "discover" by seeing it in action in other MRs. Still, for those who want it its pretty much already available, you just need to set two labels instead of one. We could also trigger a "triage run" when a label is set, but those happen pretty regularly anyway.

> > A PR may already be in review, or the author may have already pinged a reviewer.
> 
> An author is only allowed to set needs_reviewer though, right? Also, Marvin could figure out existing reviwers and reviews.

They can set everything except `needs_merger`. In that case they could/should set `awaiting_reviewer` or `awaiting_changes`.

I'll close this as "won't fix" for now. If you want you can open separate issues focused on

- setting a default status and
- triggering triage on manual label change.

Those would be fairly low priority for me though. The first one in particular would require a bit of time and is only relevant in the current experimental setting. The second one would probably be an easy enough change, but with little practical impact.

Contributions are also very welcome, but you would have to set up your own test instance. Thanks again for the feedback!
